pkg/maven/maven.go

```diff
@@ -162,6 +163,57 @@ func InstallMavenArtifacts(options *EvaluateOptions, utils Utils) error {
 	return doInstallMavenArtifacts(options, utils)
 }
 
+func FindArtifacts(options *EvaluateOptions, utils Utils) (piperenv.Artifacts, error) {
+	var artifacts piperenv.Artifacts
+
+	pomFiles, err := utils.Glob(filepath.Join("**", "pom.xml"))
+	if err != nil {
+		return nil, err
+	}
+
+	for _, pomFile := range pomFiles {
+		dir := filepath.Dir(pomFile)
+
+		finalName, err := Evaluate(options, "project.build.finalName", utils)
+		if err != nil {
+			return nil, err
+		}
+		if finalName == "" {
+			log.Entry().Warnf("project.build.finalName is empty for module '%s', can not write artifacts to common pipeline environment", pomFile)
+			continue
+		}
+
+		jarFileName := jarFile(dir, finalName)
+		warFileName := warFile(dir, finalName)
+		classesJarFileName := classesJarFile(dir, finalName)
+
+		jarExists, _ := utils.FileExists(jarFileName)
+		warExists, _ := utils.FileExists(warFileName)
+		classesJarExists, _ := utils.FileExists(classesJarFileName)
+
+		if jarExists {
+			artifacts = append(artifacts, piperenv.Artifact{
+				Kind: "java:jar",
+				Path: jarFileName,
+			})
+		}
+		if warExists {
+			artifacts = append(artifacts, piperenv.Artifact{
+				Kind: "java:war",
+				Path: warFileName,
+			})
+		}
+		if classesJarExists {
+			artifacts = append(artifacts, piperenv.Artifact{
+				Kind: "java:classes-jar",
+				Path: classesJarFileName,
+			})
+		}
```

Comment:
If we need to express both `java` and `war`/`jar`/`classes-jar`, would it be worth directly going for generic tags? E.g.
```suggestion
		if jarExists {
			artifacts = append(artifacts, piperenv.Artifact{
				Tags: ["java, "jar"],
				Path: jarFileName,
			})
		}
		if warExists {
			artifacts = append(artifacts, piperenv.Artifact{
				Tags: ["java, "war"],
				Path: warFileName,
			})
		}
		if classesJarExists {
			artifacts = append(artifacts, piperenv.Artifact{
				Tags: ["java, "classes-jar"],
				Path: classesJarFileName,
			})
		}
```

Reply:
We decided to exclude tags/labels from this PR as of now, since we don't have a particular use case for them. As soon as we get better insights into the usage, we can always add it on top